hicommonwealth/commonwealth
# Launch farcaster contest on UI

## Link to Issue
Closes: #9674 

## Description of Changes
-  Refactored "enable stake step". Initially, it was designed to work only from create community flow and enabling namespace & stake was possible in one step. Now, we need only namespace enablement AND namespace + stake enablement. Additionally, it is available from 4 different flow (create community, manage community, create topic, create contest). Refactoring was necessary to simplify all those conditions. 

Current flows:

![image](https://github.com/user-attachments/assets/aca60d17-4086-4e28-a9b7-d8cb8915c88e)

- adjusted multiple places in the FE app to display farcaster related UI
- made launching farcaster contest possible from the UI

## Test Plan
- FLAG_FARCASTER_CONTEST=true
- try to create contest 
- contest type selection screen should appear
- select farcaster contest
- if you dont have namespace enabled, you should see namespace enablement flow visible
- otherwise you will see launch contest form adjusted to the farcaster (no topic selection, additional description field)
- once successfully launched, you can see "copy farcaster contest" button in contest card => it should copy url to the clipboard
- this PR ends at this point, next things to test would be in upcoming PRs

## Deployment Plan

1. for now FLAG_FARCASTER_CONTEST flag should be false

## Other Considerations

- This PR is about launching farcaster contest from our up. There still things to do to make sure that the whole flow works (publishing it on farcaster, voting, collecting prizes etc)
